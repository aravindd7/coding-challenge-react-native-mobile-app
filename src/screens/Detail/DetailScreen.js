import React from "react";

import { WebView } from "react-native";

class DetailScreen extends React.PureComponent {
  static navigationOptions = ({ navigation }) => ({
    title: navigation.state.params.item.data.title
  });

  render() {
    const { item } = this.props.navigation.state.params; // Get info from Navigation
    const { url } = item.data;

    return <WebView source={{ uri: url }} />;
  }
}

export default DetailScreen;
