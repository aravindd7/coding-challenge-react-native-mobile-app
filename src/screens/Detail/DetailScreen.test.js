import React from "react";

import ShallowRenderer from "react-test-renderer/shallow"; // ES6

import DetailScreen from "./DetailScreen";

const props = {
  navigation: {
    state: {
      params: {
        title: "title",
        url: "http://example.com"
      }
    }
  }
};

it("match on Shallow mode", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<DetailScreen {...props} />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

it("it should apply navigation title", () => {
  const navigationOptions = DetailScreen.navigationOptions(props);
  expect(navigationOptions).toMatchSnapshot();
});
