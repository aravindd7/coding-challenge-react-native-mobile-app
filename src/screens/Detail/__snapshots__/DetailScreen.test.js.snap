// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should apply navigation title 1`] = `
Object {
  "title": "title",
}
`;

exports[`match on Shallow mode 1`] = `
<WebView
  scalesPageToFit={true}
  source={
    Object {
      "uri": "http://example.com",
    }
  }
/>
`;
